### DISCUSSION TITLE
[FEATURE REQUEST] When `backup_mode: minion` is set, allow states to individually skip file state backups

### DISCUSSION THREAD

**Discussion by xcorvis** [saltstack/salt#67434] [Feature Requests]
**Problem**
I have my minions set to use `backup_mode: minion`, but there are some files I do not want backups of - large files, or ones that change often or have no value to restore later. 

The documentation is unclear to me, and I assumed that I could specify `backup: ''` in states like file.managed in order to disable the backup of a particular file. However, this doesn't appear to work.

**Solution**
When `backup_mode: minion` is set in the minion config, states that do file backups (file.managed) should allow `backup: ''` to disable backups.

A more friendly value than the empty string may be preferable.  Also "backup" is used differently in several file states. `backup_mode: none` would be more clear to me.

**Alternatives**
It is certainly possible to leave backup_minion blank and specify `backup: minion` for all the files I want backups for, but if there are a lot I risk missing some. IMO it's safer to use the global backup setting to ensure I catch all the critical things, and then exempt non-critical ones.

**Additional context**
The backup_mode documentation is possibly misleading, and is missing at least one function (list_backup_dir)
https://docs.saltstack.com/en/latest/ref/states/backup_mode.html

This might actually be a bug, but again, it wasn't clear to me from the documentation if this was expected to work.

**Comment by doubletwist13** [comment]
FWIW, I think setting "backup: disable"   (or any actual text) works, I think just as an unintended consequence of how it's coded. 
Obviously it'll be much better to have it properly coded and documented rather than depending on an unintended outcome.
